src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -0,0 +1,45 @@
+package core.basesyntax;
+
+import static core.basesyntax.Name.RECTANGLE;
+
+public class Rectangle extends Figure {
+    private int length;
+    private int width;
+
+    public Rectangle(int length, int width) {
+        this.length = length;
+        this.width = width;
+        this.name = getName();
+        this.area = getArea();
+        this.color = getColor();
+    }
```

Comment:
1) Color should be common property hold by parent class
2) area is result of some calculations of other properties so no need to save it separately
```suggestion
    public Rectangle(int length, int width, Color color) {
        super(color)
        this.length = length;
        this.width = width;
    }
```
Similar changes required for all other figures